float distance_origin
float distance_traveled